### PR TITLE
Collect templates from TemplateFlow instead of package data

### LINKED ISSUE #48: Drop internal copies of templates in favor of re-orienting TemplateFlow templates
## Summary

The internal MNI152NLin2009cAsym and MNIInfant templates in QSIPrep are just re-oriented LPS versions of the ones from TemplateFlow, so we can start grabbing them from TemplateFlow and re-orienting them within QSIPrep.

Related to https://github.com/PennLINC/qsiprep/issues/803.

### PR DESCRIPTION
**Pull request by tsalo**
Closes #48.

## Changes proposed in this pull request

- Remove templates that are only used by QSIPrep.
- Download necessary templates for TemplateFlow and reorient them to LPS with 3dResample.
- Remove remaining templates that were used by QSIRecon.

### CODE REVIEW

**Comment by tsalo** [comment]
We'll want to cache the downloaded templates in the build image so users aren't downloading them at runtime.

File: qsirecon/workflows/recon/anatomical.py [hunk 1]
```diff
@@ -519,13 +519,24 @@ def _get_status():
...
 
+    # XXX: This is a temporary solution until QSIRecon supports flexible output spaces.
+    get_template = pe.Node(
+        GetTemplate(
+            template_name="MNI152NLin2009cAsym" if not config.workflow.infant else "MNIInfant",
+        ),
+        name="get_template",
+    )
+    reorient_to_lps = pe.Node(
+        afni.Resample(orientation="LPS", outputtype="NIFTI_GZ"),
```
Comment (tsalo) [inline qsirecon/workflows/recon/anatomical.py:530]: @mattcieslak on Slack you said RAI. Should I change this?

File: qsirecon/interfaces/anatomical.py [hunk 2]
```diff
@@ -137,15 +134,6 @@ def _run_interface(self, runtime):
...
         )
-        if not self.inputs.infant_mode:
-            self._results["template_image"] = pkgrf(
-                "qsirecon",
-                "data/mni_1mm_t1w_lps_brain.nii.gz",
-            )
-        else:
-            self._results["template_image"] = pkgrf(
-                "qsirecon", "data/mni_1mm_t1w_lps_brain_infant.nii.gz"
-            )
```
Comment (tsalo) [inline qsirecon/interfaces/anatomical.py:148]: This step _was_ using skullstripped versions of the templates, but TemplateFlow doesn't have a skullstripped version of the MNIInfant templates (although it has a brain mask I could apply to the non-skullstripped template). @mattcieslak do you think that will be a problem?

Comment (mattcieslak) [reply qsirecon/interfaces/anatomical.py:148]: We definitely want to use skull-stripped images for registration. Maybe we could apply the mask and then reorient?

File: qsirecon/interfaces/anatomical.py [hunk 3]
```diff
@@ -344,3 +332,38 @@ def _run_interface(self, runtime):
...
+    input_spec = _GetTemplateInputSpec
+    output_spec = _GetTemplateOutputSpec
+
+    def _run_interface(self, runtime):
+        from templateflow.api import get as get_template
+
+        template_file = str(
+            get_template(
+                self.inputs.template_name,
+                cohort=[None, "1"],
```
[the remark this answers is not in the document]
Comment (mattcieslak) [reply qsirecon/interfaces/anatomical.py:359]: I think I was using cohort 2, but maybe we want to make this more customizable? hbcd will want to use different infant templates for V01 and V02

File: qsirecon/interfaces/anatomical.py [hunk 4]
```diff
@@ -65,19 +64,17 @@ class QSIReconAnatomicalIngressOutputSpec(TraitedSpec):
     t1_2_mni_reverse_transform = File()
     # sub-1_from-T1w_to-MNI152NLin2009cAsym_mode-image_xfm.h5
     t1_2_mni_forward_transform = File()
-    # Generic: what template was used?
-    template_image = File()
 
 
-class QSIReconAnatomicalIngress(SimpleInterface):
+class QSIPrepAnatomicalIngress(SimpleInterface):
     """Get only the useful files from a QSIRecon anatomical output.
```
Comment (mattcieslak) [inline qsirecon/interfaces/anatomical.py:70]:
```suggestion
    """Get only the useful files from a QSIPrep anatomical output.
```

File: qsirecon/workflows/recon/anatomical.py [hunk 5]
```diff
@@ -269,18 +269,18 @@ def gather_qsiprep_anatomical_data(subject_id):
         "has_freesurfer": False,
     }
     recon_input_dir = config.execution.bids_dir
-    # Check to see if we have a T1w preprocessed by QSIRecon
+    # Check to see if we have a T1w preprocessed by QSIPrep
```
Comment (mattcieslak) [inline qsirecon/workflows/recon/anatomical.py:272]: Note to self: open an issue that handles when there is a T2w available

File: qsirecon/workflows/recon/anatomical.py [hunk 6]
```diff
@@ -807,46 +818,6 @@ def _get_resampled(atlas_configs, atlas_name, to_retrieve):
     return atlas_configs[atlas_name][to_retrieve]
 
 
-def get_t1w_registration_node(infant_mode, sloppy, omp_nthreads):
```
Comment (mattcieslak) [inline qsirecon/workflows/recon/anatomical.py:810]: Does this not get used anywhere? I think I added it as part of a plan to enable spatial normalization during the recon workflows if one was missing but may have not used it anywhere

**Review by mattcieslak — CHANGES_REQUESTED** [review]

File: qsirecon/workflows/recon/anatomical.py (hunk 6 above)
Comment (tsalo) [reply qsirecon/workflows/recon/anatomical.py:810]: It gets called in some commented-out code, but I think we'll take a different tack with the spatial normalization (i.e., output-spaces and whatnot).

File: qsirecon/interfaces/anatomical.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (tsalo) [reply qsirecon/interfaces/anatomical.py:359]: I definitely want to make it more customizable, but I think that falls under #44.

**Comment by tsalo** [comment]
@mattcieslak does this look good to merge now?

**Pull request merged**